nixos/tests/activation/etc-overlay-immutable.nix

```diff
@@ -23,6 +23,15 @@
     specialisation.new-generation.configuration = {
       environment.etc."newgen".text = "newgen";
     };
+
+    assertions = lib.mkIf (lib.length (lib.attrNames config.specialisation) > 0) [
```

Comment:
This `mkIf` is needed because otherwise this assertion is also added to the specialisations, which leads to an error because they don't have any specialisations themselves.

Reply:
Can't we just check this during runtime of the test? I.e. compare their sha256sum?

I think that would be a little cleaner.

Reply:
Done.